### ISSUE TITLE
Debugging doesn't work anymore in C++ using g++

### ISSUE THREAD

**Issue by aaryan-rampal** [microsoft/vscode-cpptools#7093]
Issue Type: <b>Bug</b>

```json
{
    "tasks": [
        {
            "type": "cppbuild",
            "label": "C/C++: g++.exe build active file",
            "command": "C:\\Users\\Owner\\MinGW\\mingw64\\bin\\g++.exe",
            "args": [
                "-g",
                "${file}",
                "-o",
                "${fileDirname}\\${fileBasenameNoExtension}.exe"
            ],
            "options": {
                "cwd": "${workspaceFolder}"
            },
            "problemMatcher": [
                "$gcc"
            ],
            "group": {
                "kind": "build",
                "isDefault": true
            },
            "detail": "Task generated by Debugger."
        }
    ],
    "version": "2.0.0"
}
```

This is my tasks.json file. When I press F5 on a C++ file, I get the option to select my environment. Then, it gives me another option to build and debug with 'g++.exe - Build and debug active file' or a default configuration. If I choose the g++.exe, it'll create a launch,json file that looks like this

```json
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "g++.exe - Build and debug active file",
            "type": "cppdbg",
            "request": "launch",
            "program": "${fileDirname}\\${fileBasenameNoExtension}.exe",
            "args": [],
            "stopAtEntry": false,
            "cwd": "C:\\Users\\Owner\\MinGW\\mingw64\\bin",
            "environment": [],
            "externalConsole": false,
            "MIMode": "gdb",
            "miDebuggerPath": "C:\\Users\\Owner\\MinGW\\mingw64\\bin\\gdb.exe",
            "setupCommands": [
                {
                    "description": "Enable pretty-printing for gdb",
                    "text": "-enable-pretty-printing",
                    "ignoreFailures": true
                }
            ],
            "preLaunchTask": "C/C++: g++.exe build active file"
        }
    ]
}
```

However, there is a task error.
```
Error: The cppbuild task detection didn't contribute a task for the following configuration:
{
    "type": "cppbuild",
    "label": "C/C++: g++.exe build active file",
    "command": "C:\\Users\\Owner\\MinGW\\mingw64\\bin\\g++.exe",
    "args": [
        "-g",
        "${file}",
        "-o",
        "${fileDirname}\\${fileBasenameNoExtension}.exe"
    ],
    "options": {
        "cwd": "${workspaceFolder}"
    },
    "problemMatcher": [
        "$gcc"
    ],
    "group": {
        "kind": "build",
        "isDefault": true
    },
    "detail": "Task generated by Debugger."
}
The task will be ignored.
Error: The cppbuild task detection didn't contribute a task for the following configuration:
{
    "type": "cppbuild",
    "label": "C/C++: g++.exe build active file",
    "command": "C:\\Users\\Owner\\MinGW\\mingw64\\bin\\g++.exe",
    "args": [
        "-g",
        "${file}",
        "-o",
        "${fileDirname}\\${fileBasenameNoExtension}.exe"
    ],
    "options": {
        "cwd": "${workspaceFolder}"
    },
    "problemMatcher": [
        "$gcc"
    ],
    "group": {
        "kind": "build",
        "isDefault": true
    },
    "detail": "Task generated by Debugger."
}
The task will be ignored.
```

I do not know why this error occurs. Previously, it worked fine. However, after a VSCode and C/C++ extension update, it doesn't automatically make the 'C/C++: g++.exe build active file' file anymore. I had to manually copy and paste it as a tasks.json file from my previous projects. However, even after all that, there is still an error. Please help

Edit 1: I do now know why the markdown is highlighting some of the launch.json file red. It is not highlighted inside of VSCode.

Extension version: 1.2.2
VS Code version: Code 1.52.1 (ea3859d4ba2f3e577a159bc91e3074c5d85c0523, 2020-12-16T16:34:46.910Z)
OS version: Windows_NT x64 10.0.19042

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-3470S CPU @ 2.90GHz (4 x 2893)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|7.89GB (2.33GB free)|
|Process Argv|--crash-reporter-id 45e587fc-d2bf-405b-ab3c-dd71d215acf2|
|Screen Reader|no|
|VM|0%|
</details>

**Comment by sean-mcmanus** [comment]
Hmm, it's working for me. That error is from VS Code (https://github.com/microsoft/vscode/blob/a699ffaee62010c4634d301da2bbdb7646b8d1da/src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts#L1798 ). I'm not sure what is going on. Do you see any error messages in Help->Toggle Developer Tools->Console?

@elahehrashedi Any ideas what could be going wrong?

**Comment by jus78help** [comment]
I followed this link (https://stackoverflow.com/questions/52461400/how-to-completely-remove-vs-code-setting) to clean reinstall VSCode and remove Code and extension settings. Now, it seems to work. However, I do remember I saw a "Devtools failed to load SourceMap" error in the developer console. I read the file path and it didn't seem like it had anything to do with this extension, so I did not think much about it.
